docs/deployment/azure/customize-deployments.md

```diff
@@ -0,0 +1,262 @@
+---
+title: Customize .NET Aspire deployments with infrastructure synthesis
+description: Learn how to customize Azure deployments using azd infra synth for production scenarios.
+ms.date: 01/08/2025
+ms.custom: devx-track-extended-azdevcli
+---
+
+# Customize .NET Aspire deployments with infrastructure synthesis
+
+The Azure Developer CLI (`azd`) provides a powerful feature called infrastructure synthesis that allows you to generate and customize the underlying infrastructure code for your .NET Aspire applications. This capability is essential for production scenarios where you need fine-grained control over Azure resources, security configurations, and deployment patterns.
+
+This article covers how to use `azd infra synth` to:
+
+> [!div class="checklist"]
+>
+> - Generate Bicep infrastructure files from your .NET Aspire app model
+> - Customize generated infrastructure for production requirements
+> - Apply security best practices to generated resources
+> - Manage infrastructure as code with proper version control
+
+[!INCLUDE [aspire-prereqs](../../includes/aspire-prereqs.md)]
+
+You will also need to have the Azure Developer CLI [installed locally](/azure/developer/azure-developer-cli/install-azd).
+
+## How infrastructure synthesis works
+
+Infrastructure synthesis in `azd` transforms your .NET Aspire app model into concrete Azure infrastructure definitions using Bicep templates. This process bridges the gap between the development-time orchestration in .NET Aspire and the production infrastructure required in Azure.
+
+When you run `azd infra synth`, the CLI:
+
+1. Analyzes your .NET Aspire app host project
+2. Identifies all resources and their dependencies
+3. Generates corresponding Azure resource definitions in Bicep
+4. Creates supporting configuration files for deployment
+
+## Enable and use infrastructure synthesis
+
+Infrastructure synthesis is currently an alpha feature that must be explicitly enabled:
+
+```azdeveloper
+azd config set alpha.infraSynth on
+```
+
+Once enabled, generate infrastructure files for your .NET Aspire project:
+
+```azdeveloper
+azd infra synth
+```
+
+This command creates an `infra` folder in your app host project directory with the following structure:
+
+```
+infra/
+├── main.bicep              # Main infrastructure entry point
+├── main.parameters.json    # Parameter values for deployment
+├── resources.bicep         # Resource definitions
+└── abbreviations.json      # Azure resource naming conventions
+```
+
+## Production considerations
+
```

Comment:
As a style preference, I don't care much for headings without at least an introduction sentence, or paragraph. Anytime I see multiple headings back-to-back without context, that's a code smell to me. Let's please add something here to introduce this section.